rfcs/proposed/iterators_pass_directly_customization/README.md

```diff
@@ -117,40 +124,55 @@ namespace user
     };
 
     template <typename It1, typename It2>
-    constexpr 
     auto is_passed_directly_in_onedpl_device_policies(const iterator_pair<It1, It2>&)
     {
-        if constexpr (oneapi::dpl::is_passed_directly_to_device_v<It1> &&
-                      oneapi::dpl::is_passed_directly_to_device_v<It2>)
-            return std::true_type{};
-        else 
-            return std::false_type{}; 
+        return std::conjunction<oneapi::dpl::is_passed_directly_to_device<It1>,
+                                oneapi::dpl::is_passed_directly_to_device<It2>>{};
     }
 } //namespace user
 ```
+
 It is also possible to write overloads without a body using an `auto` trailing return type. The following is an
 alternative written in this way for the previous example:
+
 ```cpp
     template <typename It1, typename It2>
-    constexpr 
+
     auto is_passed_directly_in_onedpl_device_policies(const iterator_pair<It1, It2>&) ->
-        std::bool_constant<(oneapi::dpl::is_passed_directly_to_device_v<It1> &&
-                            oneapi::dpl::is_passed_directly_to_device_v<It2>)>;
+        std::conjunction<oneapi::dpl::is_passed_directly_to_device<It1>,
+                         oneapi::dpl::is_passed_directly_to_device<It2>>>;
+```
+Finally, it is possible to write overloads using the "hidden friend" idiom as functions with or without a body inside
+the scope of the iterator definition itself. This option may be preferred when a user wants to ensure that this "passed
+directly" trait of their iterator is coupled tightly with the definition itself for maintenance. It also has the
+advantage of only being making available for ADL the combinations of template arguments for this iterator which are
+explicitly instantiated in the code already.
+
+```cpp
+    template <typename It1, typename It2>
+    struct iterator_pair
+    {
+        It1 first;
+        It2 second;
+        auto is_passed_directly_in_onedpl_device_policies(const iterator_pair&) ->
+            std::conjunction<oneapi::dpl::is_passed_directly_to_device<It1>,
+                             oneapi::dpl::is_passed_directly_to_device<It2>>;
```

Comment:
This is not a "hidden friend" function but a method.

Reply:
good catch.  I left out the friend keyword.  I'll fix in a second.